### PR TITLE
Added register and funcionality

### PR DESCRIPTION
**Pull request by rlajous**
## Summary 

- Added register in home

## Screenshot 

vue1

## Trello Card

https://trello.com/c/hEW59uiL/22-register-layout-exercise

### CODE REVIEW

File: wolox-books/src/views/Home.vue [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
-      <label for="password" class="label">Password</label>
-      <input type="password" name="firstname" id="password">
-      <button>Sign Up</button>
-    </form>
-    <a href="/">Login</a>
-  </div>
+<template lang='pug'>
+#register-form.home
+  form.register-form(@submit.prevent="onSubmit")
+    label.label(for="firstName") First Name
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Use pipe notation

File: wolox-books/src/views/Home.vue [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
+    input#firstName(type="text",name="firstName",v-model="firstName")
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Separate the attributes with a space, not need for the comma

Comment (mcallegari10) [reply wolox-books/src/views/Home.vue]: Also, no need for the id

File: wolox-books/src/views/Home.vue [hunk 3]
```diff
@@ -1,22 +1,40 @@
...
-      <label for="email" class="label">Email</label>
-      <input type="text" name="firstname" id="email">
-      <label for="password" class="label">Password</label>
-      <input type="password" name="firstname" id="password">
-      <button>Sign Up</button>
-    </form>
-    <a href="/">Login</a>
-  </div>
+<template lang='pug'>
+#register-form.home
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: No need for the id

File: wolox-books/src/views/Home.vue [hunk 4]
```diff
@@ -1,22 +1,40 @@
...
+  form.register-form(@submit.prevent="onSubmit")
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Remember to add a `novalidate` to the form so the errors are custom and not HTML based

File: wolox-books/src/views/Home.vue [hunk 5]
```diff
@@ -1,22 +1,40 @@
...
+    label.label(for="firstName") First Name
+    input#firstName(type="text",name="firstName",v-model="firstName")
+    label.label(for="lastName") Last Name
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Pipe notation

File: wolox-books/src/views/Home.vue [hunk 6]
```diff
@@ -1,22 +1,40 @@
...
+    input#lastName(type="text",name="lastName",v-model="lastName")
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Separate with spaces and no need for the id

File: wolox-books/src/views/Home.vue [hunk 7]
```diff
@@ -1,22 +1,40 @@
...
 
 <script>
-
+module.exports = {
+  data() {
+    return {
+      firstName: "",
+      lastName: "",
+      email: "",
+      password: ""
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Change all of this to single quotes

**Review by mcallegari10 — CHANGES_REQUESTED** [review]

File: wolox-books/src/views/Home.vue (hunk 1 above)
Comment (rlajous) [reply wolox-books/src/views/Home.vue]: it is fixed in the other pr

File: wolox-books/src/views/Home.vue (hunk 2 above)
Comment (rlajous) [reply wolox-books/src/views/Home.vue]: it is fixed in the other pr

File: wolox-books/src/views/Home.vue (hunk 5 above)
Comment (rlajous) [reply wolox-books/src/views/Home.vue]: it is fixed in the other pr

File: wolox-books/src/views/Home.vue (hunk 2 above)
Comment (rlajous) [reply wolox-books/src/views/Home.vue]: The id is for the label

File: wolox-books/src/views/Home.vue (hunk 7 above)
Comment (rlajous) [reply wolox-books/src/views/Home.vue]: eslint is autofixing this

File: wolox-books/.gitignore [hunk 8]
```diff
@@ -1,6 +1,7 @@
 .DS_Store
 node_modules
 /dist
+/package-lock.json
```
Comment (FrankIglesias) [inline wolox-books/.gitignore]: the package-lock should not be gitignored

File: wolox-books/src/views/Home.vue [hunk 9]
```diff
@@ -1,22 +1,63 @@
...
+        last_name: this.lastName,
+        email: this.email,
+        password: this.password
+      };
+      console.log(data);
+      this.$v.$touch();
+      if (this.$v.$invalid) {
+        this.submitStatus = "ERROR";
+      } else {
+        // do your submit logic here
```
Comment (FrankIglesias) [inline wolox-books/src/views/Home.vue]: remove comment and semi colon of the file

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: wolox-books/src/views/Home.vue [hunk 10]
```diff
@@ -1,25 +1,65 @@
...
+      const data = {
+        first_name: this.firstName,
+        last_name: this.lastName,
+        email: this.email,
+        password: this.password
+      }
+      console.log(data)
+      this.$v.$touch()
+      if (this.$v.$invalid) {
+        this.submitStatus = 'ERROR'
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Remember to change this status attribute :stuck_out_tongue_closed_eyes:

File: wolox-books/src/views/Home.vue [hunk 11]
```diff
@@ -1,25 +1,65 @@
...
+      } else {
+        this.submitStatus = 'PENDING'
+        setTimeout(() => {
```
Comment (mcallegari10) [inline wolox-books/src/views/Home.vue]: Not need for this time out

File: wolox-books/.env [hunk 12]
```diff
@@ -0,0 +1 @@
+VUE_APP_API_BASE_URL="https://wbooks-api-stage.herokuapp.com/api/v1"
```
Comment (FrankIglesias) [inline wolox-books/.env]: you  must gitignore this

File: wolox-books/src/views/Home.vue [hunk 13]
```diff
@@ -1,25 +1,58 @@
...
+  },
+  methods: {
+    onSubmit () {
+      const data = {
+        first_name: this.firstName,
+        last_name: this.lastName,
+        email: this.email,
+        password: this.password
+      }
+      console.log(data)
```
Comment (FrankIglesias) [inline wolox-books/src/views/Home.vue:42]: remove the ocnsole log

File: wolox-books/src/views/Home.vue [hunk 14]
```diff
@@ -1,25 +1,58 @@
...
+  validations: {
+    password: {
+      required
+    }
+  }
+}
 </script>
 
-<style lang="scss">
+<style lang='scss'>
```
Comment (FrankIglesias) [inline wolox-books/src/views/Home.vue]: use scoped styles

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: wolox-books/src/views/Home.vue (hunk 13 above)
Comment (rlajous) [reply wolox-books/src/views/Home.vue:42]: The card asked for this

**Review by mcallegari10 — APPROVED** [review]

**Pull request merged**